### PR TITLE
Add debugger config

### PR DESCRIPTION
**Pull request by jpbempel**
# What Does This Do

Add debugger config options

# Motivation

Config options required by upcoming Live Debugger product

# Additional Notes

### CODE REVIEW

File: internal-api/src/main/java/datadog/trace/api/Config.java [hunk 1]
```diff
@@ -457,6 +483,22 @@ public class Config {
...
+  private final boolean debuggerClassFileDumpEnabled;
+  private final int debuggerPollInterval;
+  private final int debuggerDiagnosticsInterval;
+  private final boolean debuggerMetricEnabled;
+  private final String debuggerProbeFileLocation;
+  private final int debuggerBatchSize;
+  private final long debuggerMaxPayloadSize;
+  private final boolean debuggerVerifyByteCode;
+  private final boolean debuggerInstrumentTheWorld;
+  private final String debuggerExcludeFile;
```
Comment (devinsba) [inline internal-api/src/main/java/datadog/trace/api/Config.java:500]: Do you want to add these fields to the toString() method too?

File: dd-trace-api/src/main/java/datadog/trace/api/ConfigDefaults.java [hunk 2]
```diff
@@ -78,6 +78,18 @@ public final class ConfigDefaults {
   static final boolean DEFAULT_CIVISIBILITY_ENABLED = false;
   static final boolean DEFAULT_CIVISIBILITY_AGENTLESS_ENABLED = false;
 
+  static final boolean DEFAULT_DEBUGGER_ENABLED = false;
+  static final int DEFAULT_DEBUGGER_UPLOAD_TIMEOUT = 30; // seconds
+  static final int DEFAULT_DEBUGGER_UPLOAD_FLUSH_INTERVAL = 0; // ms, 0 = dynamic
```
Comment (devinsba) [inline dd-trace-api/src/main/java/datadog/trace/api/ConfigDefaults.java:83]: Is there a reason to have this one config in milliseconds while all the others are seconds?

**Review by devinsba — APPROVED** [review]

File: internal-api/src/main/java/datadog/trace/api/Config.java (hunk 1 above)
Comment (jpbempel) [reply internal-api/src/main/java/datadog/trace/api/Config.java:500]: ah true, thanks!

File: dd-trace-api/src/main/java/datadog/trace/api/ConfigDefaults.java (hunk 2 above)
Comment (jpbempel) [reply dd-trace-api/src/main/java/datadog/trace/api/ConfigDefaults.java:83]: yes, flushing interval will be no more than 2 seconds here, so having ms precision here makes sense

**Pull request merged**